.github/workflows/build-test-n-publish.yml

```
@@ -373,6 +373,9 @@ jobs:
       run: python -m tox -p auto --parallel-live -vvvv --notest
     - name: Run tests
       run: python -m tox -p auto --parallel-live -vvvv
+    - name: Setup tmate session
+      if: failure()
```

This can prevent pull request events from polluting the upstream:
```suggestion
      if: failure() && github.event_name == 'push'
```